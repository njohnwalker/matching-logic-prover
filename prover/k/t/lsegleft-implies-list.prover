claim \implies( \and( lsegleft( variable("H"), variable("X"), variable("Y"), variable("F"), .Patterns) 
                  , list( variable("H"), variable("Y"), variable("G"), .Patterns)
                  , \equals( variable("K"), union(variable("F"), variable("G")))
                  , disjoint(variable("F"), variable("G"))
                  , .Patterns
                  )
            , \and( list( variable("H"), variable("X"), variable("K"), .Patterns), .Patterns )
            )
strategy search-bound(4)
