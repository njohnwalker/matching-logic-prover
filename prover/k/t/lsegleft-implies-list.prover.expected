<prover>
  <goal>
    <id>
      0
    </id>
    <active>
      true
    </active>
    <parent>
      -1
    </parent>
    <k>
      \implies ( \and ( lsegleft ( variable ( "H" ) , variable ( "X" ) , variable ( "Y" ) , variable ( "F" ) , .Patterns ) , list ( variable ( "H" ) , variable ( "Y" ) , variable ( "G" ) , .Patterns ) , \equals ( variable ( "K" ) , union ( variable ( "F" ) , variable ( "G" ) ) ) , disjoint ( variable ( "F" ) , variable ( "G" ) ) , .Patterns ) , \and ( list ( variable ( "H" ) , variable ( "X" ) , variable ( "K" ) , .Patterns ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
      .
    </trace>
  </goal>
</prover>
